crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -87,46 +73,63 @@ impl EthereumSettlementClient {
         let wallet_address = signer.address();
         let wallet = EthereumWallet::from(signer);
 
-        let provider = Arc::new(
+        // provider without wallet
+        let provider = Arc::new(ProviderBuilder::new().on_http(settlement_cfg.rpc_url.clone()));
+
+        // provider with wallet
+        let filler_provider = Arc::new(
             ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
         );
+
         let core_contract_client = StarknetValidityContractClient::new(
             Address::from_str(&settlement_cfg.core_contract_address)
                 .expect("Failed to convert the validity contract address.")
                 .0
                 .into(),
-            provider.clone(),
+            filler_provider,
         );
 
-        EthereumSettlementClient { provider, core_contract_client, wallet, wallet_address }
+        EthereumSettlementClient {
+            provider,
+            core_contract_client,
+            wallet,
+            wallet_address,
+            #[cfg(test)]
+            impersonate_account: None,
+        }
     }
 
     #[cfg(test)]
-    pub fn with_test_settings(settings: &impl SettingsProvider, provider: RootProvider<Http<Client>>) -> Self {
-        use tests::{SHOULD_IMPERSONATE_ACCOUNT, TEST_DUMMY_CONTRACT_ADDRESS};
-        let settlement_cfg: EthereumSettlementConfig = settings.get_settings(SETTLEMENT_SETTINGS_NAME).unwrap();
-
+    pub fn with_test_settings(
+        provider: RootProvider<Http<Client>>,
+        core_contract_address: Option<Address>,
+        rpc_url: Url,
+        impersonate_account: Option<Address>,
+    ) -> Self {
         let private_key = get_env_var_or_panic(ENV_PRIVATE_KEY);
         let signer: PrivateKeySigner = private_key.parse().expect("Failed to parse private key");
         let wallet_address = signer.address();
         let wallet = EthereumWallet::from(signer);
 
-        let fill_provider = Arc::new(
-            ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(settlement_cfg.rpc_url),
-        );
+        let fill_provider =
+            Arc::new(ProviderBuilder::new().with_recommended_fillers().wallet(wallet.clone()).on_http(rpc_url));
 
-        let core_contract_address = if *SHOULD_IMPERSONATE_ACCOUNT {
-            &settlement_cfg.core_contract_address
+        let core_contract_address = if let Some(core_contract_address) = core_contract_address {
+            core_contract_address.clone()
         } else {
-            &*TEST_DUMMY_CONTRACT_ADDRESS
+            // dummy address
+            Address::from_str("0x0").unwrap()
```

Comment:
If the dummy contract Address is `0x0` how will the dummy transaction be made ?
Moreover Why are we not sending the DummyContractAddress in mod.rs line 143 ? and sending None instead.